### PR TITLE
Add workflow for changelog

### PR DESCRIPTION
**Pull request by owaiskazi19**
### Description
Add workflow for changelog

### Issues Resolved
Closes https://github.com/opensearch-project/flow-framework/issues/437

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -78,6 +79,24 @@ Signed-off-by: Jane Smith <jane.smith@email.com>
 ```
 You may type this line on your own when writing your commit messages. However, if your user.name and user.email are set in your git configs, you can use `-s` or `--signoff` to add the `Signed-off-by` line to the end of the commit message.
 
+## Changelog
+
+Flow framework maintains version specific changelog by enforcing a change to the ongoing [CHANGELOG](CHANGELOG.md) file adhering to the [Keep A Changelog](https://keepachangelog.com/en/1.0.0/) format.
```
Comment (dbwiddis) [inline CONTRIBUTING.md:84]: Nit 1: "Flow Framework".  I think. Feel free to disagree.
Nit 2: I think the version is now 1.1.0 (also link elsewhere).

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+### Deprecated
+
+### Removed
+
+### Fixed
+
+### Security
+
+
+[Unreleased]: https://github.com/opensearch-project/flow-framework/compare/2.x...HEAD
```
Comment (dbwiddis) [inline CHANGELOG.md:19]: This seems like a link to a bunch of commits/diffs which is explicitly what a changelog is NOT supposed to be.  Is this used elsewhere?

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+CHANGELOG
+Inspired from [Keep a Changelog](https://keepachangelog.com/en/1.0.0/)
+
```
Comment (dbwiddis) [inline CHANGELOG.md:3]: We probably need a "2.12.0 Initial Release" line in this file somewhere. Not sure exactly where.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -78,6 +79,24 @@ Signed-off-by: Jane Smith <jane.smith@email.com>
...
+
+Briefly, the changes are curated by version, with the changes to the main branch added chronologically to `Unreleased` version. Further, each version has corresponding sections which list out the category of the change - `Added`, `Changed`, `Deprecated`, `Removed`, `Fixed`, `Security`.
+
+
```
Comment (dbwiddis) [inline CONTRIBUTING.md:88]: Really low-level nit: I see two linebreaks here. I'm surprised spotless didn't fix this.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -78,6 +79,24 @@ Signed-off-by: Jane Smith <jane.smith@email.com>
...
+### How to add my changes to [CHANGELOG](CHANGELOG.md)?
+
+As a contributor, you must ensure that every pull request has the changes listed out within the corresponding version and appropriate section of [CHANGELOG](CHANGELOG.md) file.
```
Comment (dbwiddis) [inline CONTRIBUTING.md:91]: I think "must" is a bit too strong.  I'd really like a bit longer paragraph here covering:
 - substantive changes _should_ be listed
 - deprecations _must_ be listed
 - trivial / typo / doc fixes do not need to be listed.   In these cases add the skip-changelog tag.

No need to invent something, I'm content with whatever is in widespread use in the org.

File: .github/workflows/changelog_verifier.yml [hunk 6]
```diff
@@ -0,0 +1,19 @@
+name: "Changelog Verifier"
+on:
+  pull_request:
+    types: [opened, edited, review_requested, synchronize, reopened, ready_for_review, labeled, unlabeled]
```
Comment (dbwiddis) [inline .github/workflows/changelog_verifier.yml:4]: Do we want to filter based on branch?   (I think probably yes.)

File: CHANGELOG.md (hunk 2 above)
Comment (owaiskazi19) [reply CHANGELOG.md:19]: [OpenSearch](https://github.com/opensearch-project/OpenSearch/blob/main/CHANGELOG.md?plain=1#L248) has it this way.

File: CHANGELOG.md (hunk 3 above)
Comment (owaiskazi19) [reply CHANGELOG.md:2]: Added on L6.

File: .github/workflows/changelog_verifier.yml (hunk 6 above)
Comment (owaiskazi19) [reply .github/workflows/changelog_verifier.yml:8]: Good idea. Added it.

File: CONTRIBUTING.md (hunk 4 above)
Comment (owaiskazi19) [reply CONTRIBUTING.md]: Well, I did :)

**Review by dbwiddis — APPROVED** [review]
LGTM, reserving the right to request changes later :)

**Review by joshpalis — APPROVED** [review]

**Pull request merged**